src/io/flutter/run/bazelTest/BazelTestFields.java

```diff
@@ -0,0 +1,252 @@
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.configurations.RuntimeConfigurationError;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.openapi.vfs.CharsetToolkit;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.jetbrains.lang.dart.sdk.DartConfigurable;
+import com.jetbrains.lang.dart.sdk.DartSdk;
+import io.flutter.FlutterBundle;
+import io.flutter.bazel.Workspace;
+import io.flutter.bazel.WorkspaceCache;
+import io.flutter.dart.DartPlugin;
+import io.flutter.run.MainFile;
+import io.flutter.run.daemon.FlutterDevice;
+import io.flutter.run.daemon.RunMode;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The fields in a Bazel test configuration.
+ */
+public class BazelTestFields {
+
+  private @Nullable String entryFile;
+  private @Nullable String launchScript;
+  //private @Nullable String additionalArgs;
+  private @Nullable String bazelTarget;
+
+  BazelTestFields() {
+  }
+
+  /**
+   * Copy constructor
+   */
+  private BazelTestFields(@NotNull BazelTestFields original) {
+    entryFile = original.entryFile;
+    launchScript = original.launchScript;
+    //additionalArgs = original.additionalArgs;
+    bazelTarget = original.bazelTarget;
+  }
+
+  /**
+   * Create non-template from template.
+   */
+  private BazelTestFields(@NotNull BazelTestFields template, Workspace w) {
+    this(template);
+    if (StringUtil.isEmptyOrSpaces(launchScript)) {
+      launchScript = w.getLaunchScript();
+      if (launchScript != null && !launchScript.startsWith("/")) {
+        launchScript = w.getRoot().getPath() + "/" + launchScript;
+      }
+    }
+  }
+
+  /**
+   * The file containing the main function that starts the Flutter app.
+   */
+  @Nullable
+  public String getEntryFile() {
+    return entryFile;
+  }
+
+  public void setEntryFile(final @Nullable String entryFile) {
+    this.entryFile = entryFile;
+  }
+
+  /**
+   * Present only for deserializing old run configs.
+   */
+  @SuppressWarnings("SameReturnValue")
+  @Deprecated
+  public String getWorkingDirectory() {
+    return null;
+  }
+
+  /**
+   * Used only for deserializing old run configs.
+   */
+  @Deprecated
+  public void setWorkingDirectory(final @Nullable String workDir) {
+    if (entryFile == null && workDir != null) {
+      entryFile = workDir + "/lib/main.dart";
+    }
+  }
+
+  @Nullable
+  public String getLaunchingScript() {
+    return launchScript;
+  }
+
+  public void setLaunchingScript(@Nullable String launchScript) {
+    this.launchScript = launchScript;
+  }
+
+  //@Nullable
+  //public String getAdditionalArgs() {
+  //  return additionalArgs;
+  //}
+  //
+  //public void setAdditionalArgs(@Nullable String additionalArgs) {
+  //  this.additionalArgs = additionalArgs;
+  //}
```

Comment:
Can we remove these?  (Or just add them in?)

Reply:
I don't know yet if we should remove or add them in.  Added TODOs.